packages/docusaurus-theme-classic/src/theme/CodeBlock/index.js

```diff
@@ -48,9 +48,13 @@ export default ({children, className: languageClassName, metastring}) => {
     };
   }, [button.current, target.current]);
 
-  const language =
+  let language =
     languageClassName && languageClassName.replace(/language-/, '');
 
+  if (language === 'undefined' && prism.defaultLanguage) {
```

Comment:
Is the 'undefined' string a default?

Reply:
I do not quite understand this, 'undefined' will be in case the language is not specified.

Reply:
```suggestion
  if (!language && prism.defaultLanguage) {
```

Following markdown
````markdown
```js live
foo()
bar()
baz()
```
````

Will yield

```js
{
  languageClassName: 'language-javascript',
  metastring: 'live=true',
  value: 'foo()\nbar()\nbaz()'
}
```

So

Following markdown
````markdown
```
foo()
bar()
baz()
```
````

Will yield

```js
{
  languageClassName: undefined,
  value: 'foo()\nbar()\nbaz()'
}
```

Reply:
Ah, I get you, but in case of using just `CodeBlock` (without `@docusaurus/theme-live-codeblock`) it will be exactly `undefined` as `undefined`, and not a string as in the case with `CodeBlockLive`

Reply:
> Ah, I get you, but in case of using just CodeBlock (without @docusaurus/theme-live-codeblock) it will be exactly undefined as string type, not a undefined type as in the case with CodeBlock with live.

Hmmm. But CodeBlock on both theme components actually receive the same `languageClassName` props though.

I just checked by removing the live codeblock on our config
```diff
- themes: ['@docusaurus/theme-live-codeblock'],
```

Still undefined though
<img width="953" alt="debug" src="https://user-images.githubusercontent.com/17883920/68186452-c1bd3e80-ffd6-11e9-8007-5a911826bc64.PNG">

Reply:
Hmm, very strange, I just tested again, and in both cases I get `undefined` as a type!
But.. OK, thanks!